.github/workflows/block_manual_pr_release.yml

```diff
@@ -0,0 +1,27 @@
+# Blocks PRs targeting the release branch that are not created by the release GHA workflow.
```

Comment:
Lets add a note here that explains why this works for our use case: Because a workflow doesn't trigger other workflows. Maybe link to the GH docs that explain this too. Basically, if GH changes their behavior at some point in future, we need to have some pointers that explain how we're relying on that behavior.

Reply:
done.